crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -357,9 +360,22 @@ impl<Wg: WireGuard, E: Backoff> State<Wg, E> {
                 .get(&server.public_key)
                 .ok_or(Error::NoStunPeer)?;
 
+            let use_ipv6 = self
```

Comment:
Ideally then we have both ipv6 and ipv4, we should try to do stun request first with ipv6, and if that fails we should try with ipv4. I like this approach as it's simpler, but I think that we cannot truly relay that ipv6 will always work. @Jauler please comment

Reply:
But how should it behave? Should it change server, how it should cooperate with exponential backoff?

Reply:
generally the retry logic can be based on two things:
* IPv6 should be preferred if it is available, due to conflict-free nature of it.
* We should not assume, that IPv6 is bullet-proof, bug-free or other impossible things :D 


So my recommendation would be:
1. Try IPv6 if it is enabled
1. If IPv6 attempt failed or it is disabled -> attempt IPv4
1. If IPv4 attempt failed, increment exponential backoff time.
1. Wait for exponential backoff to pass
1. Go to step 1

This way we still have exponential backoff increasing and increasing time between retries. We do attempt to use both IPv6 and IPv4 (meaning that in case we have some bug preventing one from working, but allowing other -> meshnet would still be operational), and we prefer IPv6 due to its conflict-free nature :).

Reply:
Ok, after a bit more discussions with @stalowyjez I realized that the approach above requires rather non-trivial amount of refactoring for WG-STUN module, which may be more risky, than then risk of WG-STUN servers not responding via IPv6 inside of the tunnel. What is more, there seems to be growing need for refactoring IPv4/IPv6 fallback logic into a common place, such that different modules would behave the same in terms of selecting IP protocol to use. But that is a task for another Jira issue and pull-request

Reply:
After some more discussions with @stalowyjez I just realized the problem. The STUN EndpointProvider is instantiated at the `start()` method. And in that context we do not yet know whether IPv6 address has been actually assigned to this meshnet node by the API. Meaning, that selecting IPv6 in that situation would be unsafe.